src/askContainer/commands/viewAllSkills.ts

```diff
@@ -146,6 +150,49 @@ export class ViewAllSkillsCommand extends AbstractCommand<void> {
         }
     }
 
+    private async checkAndSetQpItems(
+        context: vscode.ExtensionContext,
+        qp: vscode.QuickPick<vscode.QuickPickItem>
+    ): Promise<void> {
+        Logger.verbose(`Calling method: ${this.commandName}.calculateTimeExceedHourly, args:`, qp);
+        const previous = context.globalState.get(VIEW_ALL_SKILLS_LAST_UPDATE_TIME) as number;
+        const now = new Date().getTime();
+
+        if (previous === undefined || (previous !== undefined && this.calculateTimeExceedHourly(previous, now, NUMBER_OF_INTERVALS))) {
+            await this.setLastUpdateTimeAndRefreshList(context, now, qp);
+            return;
+        }
+        await this.setQpItems(context, qp);
+    }
+
+    private calculateTimeExceedHourly(previous: number, now: number, limit: number) {
+        Logger.verbose(`Calling method: ${this.commandName}.calculateTimeExceedHourly, args:`, previous, now, limit);
+        return Math.abs(now - previous) / HOUR >= limit;
+    }
+
+    private async setLastUpdateTimeAndRefreshList(context: vscode.ExtensionContext, now: number, qp: vscode.QuickPick<vscode.QuickPickItem>) {
+        Logger.verbose(`Calling method: ${this.commandName}.setLastUpdateTimeAndRefreshList, args:`, now, qp);
+        void context.globalState.update(VIEW_ALL_SKILLS_LAST_UPDATE_TIME, now);
+        clearCachedSkills(context);
+        await this.setQpItems(context, qp);
+    }
+
+    private async checkSkillExist(context: CommandContext, qp: vscode.QuickPick<vscode.QuickPickItem>): Promise<void> {
+        Logger.verbose(`Calling method: ${this.commandName}.checkSkillExist`);
+        const skillInfo = this.skillInfoMap.get(qp.activeItems[0].label)!;
+        const skillName = getSkillNameFromLocales(skillInfo.data.skillSummary.nameByLocale!);
+        try {
+            await getSkillMetadata(skillInfo.data.skillSummary.skillId!, SKILL.STAGE.DEVELOPMENT, context.extensionContext);
+        } catch (error) {
+            // reset cached skills to automatically refresh the list next time
+            this.setLastUpdateTimeAndRefreshList(context.extensionContext, new Date().getTime(), qp);
+            if (error.statusCode === 404) {
+                throw loggableAskError(`The skill '${skillName}' was not found. Please check if the skill exists in the Alexa developer console`, undefined, true);
+            }
+            throw loggableAskError(error, undefined, true);
```

Comment:
Couple suggestions here : 
- Rather than refreshing the list in the `catch` clause here (and introduce confusion to anyone else using this method for just checking if the skill exist), I suggest throw the error and handle it in the quick pick set up. 
- The current behavior shows and error and automatically refreshes the quick pick, without letting users acknowledge about the issue. I think it is mention something like `The skill '${skillName}' was not found. Please check if the skill exists in the Alexa developer console. Clicking OK refreshes the skill list` on the error message, provide an `OK` button on the message, and regardless of the button click, refresh the list. This will prevent confusion and will navigate developers properly.
- Provide the developer console hyper link in the error message, so that developers can directly jump to the console and check if the skill exists or not.

Reply:
Nice suggestions, updated it.